src/qforte/expansions/excited_state_algorithms.py

```diff
@@ -0,0 +1,74 @@
+"""
+Algorithms to do excited states based on expansions.  (E.g. q-sc-EOM)
+"""
+import qforte
+import numpy as np
+
+
+def q_sc_eom(H, U_ref, U_manifold, ops_to_compute = []):
+    """
+    Quantum, self-consistent equation-of-motion method from Asthana et. al.
+
+    H is the JW-transformed Hamiltonian.
+    U_ref is the VQE ground state circuit (or some other state not to be included in the manifold)
+    U_manifold is a list of unitaries to be enacted on |0> to generate U_vqe|i> for each |i>.
+
+    ops_to_compute is a list of JW-transformed operators that we want an array in the basis of the ground and excited states.
+    """
+    N_qb = H.num_qubits()
+    myQC = qforte.Computer(N_qb)
+    myQC.apply_circuit(U_ref)
+    E0 = myQC.direct_op_exp_val(H).real  
+    Ek, A = ritz_eigh(H, U_manifold, verbose = False)
+    Es = [E0] + Ek
+    
+    print("q-sc-EOM:")
+    print("*"*34)
+    print(f"State:          Energy (Eh)")
+    print(f"    0{E0:35.16f}")
+    for i in range(0, len(Ek)):
+        print(f"{(i+1):5}{Ek[i]:35.16f}")
+    
+    op_mats = []
+    if len(ops_to_compute) > 0:
+        #Add the reference state with coefficient 1.
+        n_states = len(Ek) + 1
+        A_plus_ref = np.zeros((n_states, n_states), dtype = "complex")
+        A_plus_ref[0, 0] = 1.0
+        A_plus_ref[1:,1:] = A
+        all_Us = [U_ref] + U_manifold
+        
+        for op in ops_to_compute:
+            op_vqe_basis = qforte.build_effective_operator(op, all_Us)
+            op_q_sc_eom_basis = A_plus_ref.T.conj()@op_vqe_basis@A_plus_ref
+            op_mats.append(op_q_sc_eom_basis.real)
+    
+    return [E0, Ek] + op_mats
```

Comment:
I want people to be able to unpack all the info in one line, especially if they don't want anything other than the energies.

tup = function()
E0, Ek = tup[0]

seems tedious.

Also all of the things being returned are basically unrelated except *maybe* E0 and Ek.  But at that point, it probably makes more sense just to combine those into one object, so that if people don't have any ops to compute, they just say

Es = function()

Reply:
I assume that Ek is a 1-dimensional numpy array of excited-state energies. What you are returning is quite inhomogeneous:
[float, numpy_array, [numpy_array_1, ..., numpy_array_n]].
I was thinking of returning something like this:
`return [E0] + list(Ek), op_mats`
When calling the function, the user can do the following:
'energies, ops = function()'